tiny_dnn/layers/convolutional_layer.h

```diff
@@ -231,7 +231,12 @@ class convolutional_layer : public feedforward_layer<Activation> {
      * @param out_data     output vectors
      **/
     void forward_propagation(const std::vector<tensor_t*>& in_data,
-                             std::vector<tensor_t*>&       out_data) override {
+                             std::vector<tensor_t*>&       out_data) override { 
+        std::cout << "BEFORE PADDED" << std::endl;
+        copy_and_pad_input(*in_data[0], cws_.prev_out_padded_);
+        *in_data[0] = cws_.prev_out_padded_;
```

Comment:
@edgarriba 
```in_data``` and ```out_data``` is shared among layers. Changing these size of vectors will affect neighbor layers which depends on their size. We should encapsulate the effect of padding/unpadding inside convolution layer.